--- src/main/java/org/apache/maven/plugin/resources/remote/ProcessRemoteResourcesMojo.java.orig	2009-09-12 06:23:41.000000000 +0800
+++ src/main/java/org/apache/maven/plugin/resources/remote/ProcessRemoteResourcesMojo.java	2010-06-02 18:15:05.000000000 +0800
@@ -47,7 +47,7 @@
 import org.apache.maven.shared.artifact.filter.collection.FilterArtifacts;
 import org.apache.maven.shared.artifact.filter.collection.GroupIdFilter;
 import org.apache.maven.shared.artifact.filter.collection.ScopeFilter;
-import org.apache.maven.shared.artifact.filter.collection.TransitivityFilter;
+import org.apache.maven.shared.artifact.filter.collection.ProjectTransitivityFilter;
 import org.apache.maven.shared.downloader.DownloadException;
 import org.apache.maven.shared.downloader.DownloadNotFoundException;
 import org.apache.maven.shared.downloader.Downloader;
@@ -609,7 +609,7 @@
             depArtifacts = project.getDependencyArtifacts();
         }
 
-        filter.addFilter( new TransitivityFilter( depArtifacts, this.excludeTransitive ) );
+        filter.addFilter( new ProjectTransitivityFilter( depArtifacts, this.excludeTransitive ) );
         filter.addFilter( new ScopeFilter( this.includeScope, this.excludeScope ) );
         filter.addFilter( new GroupIdFilter( this.includeGroupIds, this.excludeGroupIds ) );
         filter.addFilter( new ArtifactIdFilter( this.includeArtifactIds, this.excludeArtifactIds ) );
